File: src/lib/tauri/store-cache.ts
import { invoke } from "@tauri-apps/api/core";

// ── Store cache (file-backed persistence) ─────────────────────────────────────

/**
 * Write a store's serialised JSON to a file in the app data directory.
 * Replaces localStorage — no size limit.
 */
export async function saveStoreCache(key: string, json: string): Promise<void> {
  return invoke("save_store_cache", { key, json });
}

/**
 * Read a previously saved store cache. Returns null if the file doesn't exist yet.
 */
export async function loadStoreCache(key: string): Promise<string | null> {
  return invoke<string | null>("load_store_cache", { key });
}

/**
 * Delete a single store cache file.
 */
export async function deleteStoreCache(key: string): Promise<void> {
  return invoke("delete_store_cache", { key });
}

/**
 * Return the size in bytes of each cache file, keyed by cache key name.
 * Used to display cache usage in Settings.
 */
export async function getStoreCacheInfo(): Promise<Record<string, number>> {
  return invoke<Record<string, number>>("get_store_cache_info");
}

/**
 * Delete all store cache files. This is the "Clear Cache" action.
 */
export async function clearAllStoreCaches(): Promise<void> {
  return invoke("clear_all_store_caches");
}

File: src-tauri/src/commands/store_cache.rs
use crate::storage::store_cache;
use std::collections::HashMap;

/// Write (or overwrite) a cache entry.
#[tauri::command]
pub fn save_store_cache(app: tauri::AppHandle, key: String, json: String) -> Result<(), String> {
    store_cache::save_cache(&app, key, json)
}

/// Read a cache entry. Returns `None` if the file does not exist.
#[tauri::command]
pub fn load_store_cache(app: tauri::AppHandle, key: String) -> Result<Option<String>, String> {
    store_cache::load_cache(&app, key)
}

/// Delete a single cache entry. No-op if it doesn't exist.
#[tauri::command]
pub fn delete_store_cache(app: tauri::AppHandle, key: String) -> Result<(), String> {
    store_cache::delete_cache(&app, key)
}

/// Return the size in bytes of each cache file, keyed by cache key name.
/// Files that cannot be stat'd are omitted.
#[tauri::command]
pub fn get_store_cache_info(app: tauri::AppHandle) -> Result<HashMap<String, u64>, String> {
    store_cache::get_info(&app)
}

/// Delete all cache files. This is the "Clear Cache" action.
#[tauri::command]
pub fn clear_all_store_caches(app: tauri::AppHandle) -> Result<(), String> {
    store_cache::clear_all(&app)
}
